### DISCUSSION TITLE
React Native Confirm Password validation

### DISCUSSION THREAD

**Discussion by ertomz** [react-hook-form/react-hook-form#3667] [Q&A] [answered]
Hello! I'm having trouble adding a validation rule and error message to my Controller for a 'Confirm Password' input section. All examples for this API seem to be in React, but I'm in React Native. Here's what I'm working with so far within the confirmPassword Controller:

`rules={{
     required: { value: true, message: 'Confirm password is required' },
     validate: (value) => (value === getValues('password'))
}}`

How can I check that password === confirmPassword in React Native, while also adding an error message like the one for `required`?

**Comment by bluebill1049** [comment, marked as answer]
pretty much this  https://codesandbox.io/s/react-hook-form-password-match-check-standard-validation-eo6en?from-embed=&file=/src/index.js but with `Controller` instead

**Reply by ertomz** [reply]
I appreciate the speedy response! When I try that out, like this, as the example shows: 

`ref={register({
     validate: value =>
     value === password.current || "The passwords do not match"
})}`

I get an warning error 
`Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?`

I've already searched for solutions to this, but nothing so far.

**Reply by bluebill1049** [reply]
`<Controller render={(props) => <YourComponent />} rules={xxx} />` you should wire up your component.

**Reply by ertomz** [reply]
Ah! It works now when I put my rules after the component, all within the Controller. Like such:

`<Controller render{(props) => <MyComponent/>} rules={{
                                validate: value =>
                                  value === password.current || "The passwords do not match"
                            }}/>`

Turns out the ref and register pieces aren't necessary? Thanks so much!

**Reply by bluebill1049** [reply]
Awesome. glad it's working out.

**Reply by gor918** [reply]
> pretty much this https://codesandbox.io/s/react-hook-form-password-match-check-standard-validation-eo6en?from-embed=&file=/src/index.js but with `Controller` instead

The "passwords do not match" functionality is not working properly after these steps:
1. Write matching passwords
2. Add one symbol on "Repeat Password field" => you will get an error.
3. Add the same symbol on "Password field" => the error does not disappear 

The expected result: validation should trigger.

Any suggestions guys?

**Reply by zbigniewstefaniuk** [reply]
@gor918 for me in React Native it didn't works as well, but I added  simple usestate
`  const [checkPassword, setCheckPassword] = useState()`

and then in code for typing password

`            
<Controler render={(props) => (
                <>
                    <TextInput
                      onChangeText={(value) => {
                        props.onChange(value);
                        setCheckPassword(value)
                      }}
                    />
                </>
              )}
/>`


and for confirming password
`
<Controller 
            rules={{
                validate: (value) => 
                  value === checkPassword || 'The passwords do not match'
              }}
/>
`

and it worked

**Reply by bluebill1049** [reply]
awesome! @zbigniewstefaniuk you can use `watch` as well, for the first password.

`const password = watch('password')`

**Reply by rusakovic** [reply]
> `<Controller render={(props) => <YourComponent />} rules={xxx} />` you should wire up your component.

Thank your for your example. Could , please, help to fix this issue?

https://user-images.githubusercontent.com/17801144/161389679-e1fc21fb-e4ef-4178-9b41-6d2930372359.mp4
